src/System.Text.Formatting/tests/CompositeFormattingTests.cs

```diff
@@ -37,5 +37,34 @@ public void CompositeFormattingFormatStrings()
 
             Assert.Equal("HelloaFF3", formatter.ToString());
         }
+
+        [Fact]
+        public void CompositeFormattingEscaping()
+        {
+            var format = "}}a {0} b {0} c {{{0}}} d {{e}} {{";
+            var pool = new ManagedBufferPool<byte>(1024);
+            var formatter = new StringFormatter(pool);
+            formatter.Format(format, 1);
+
+            Assert.Equal(string.Format(format, 1), formatter.ToString());
```

Comment:
string.Format uses the "current culture", not invariant culture. I think the test should use: string.Format(CultureInfo.InvariantCulture, format, 1)